File: programmers/level2/42883.js
//https://school.programmers.co.kr/learn/courses/30/lessons/42883
//큰 수 만들기

function solution(number, k) {
  let answer = '';
  const splited = number.split('');

  function joinArr(fixedStr, idx) {
    // console.log('joinArr fixedStr', fixedStr);
    if (fixedStr.length === splited.length - k) {
      // console.log('===answer', answer);
      return (answer = String(Math.max(Number(answer), Number(fixedStr))));
    }

    for (let i = idx + 1; i < splited.length; i++) {
      joinArr(fixedStr + splited[i], i);
    }
  }

  for (let i = 0; i < splited.length; i++) {
    // console.log('===i', i);
    for (let j = i + 1; j < splited.length; j++) {
      joinArr(splited[i] + splited[j], j);
    }
  }

  // console.log('answer', answer);
  return answer;
}

// console.log(solution('1924', 2)); //"94"
// console.log(solution('1231234', 3)); //"3234"
console.log(solution('4177252841', 4)); //"775841"

File: programmers/python/level2/42883.py
def solution(number, k):
  stack = [number[0]]
  for num in number[1:]:
    while stack and stack[-1]<num and k>0:
      k-=1
      stack.pop()
    stack.append(num)
  if k>0:
    stack=stack[:-k]

  return "".join(stack)

# print(solution("1924",	2))
# print(solution("1231234",	3))
# print(solution("4177252841",	4))
# print(solution("11111",	4))
# print(solution("1080201",	4))
# print(solution("1120074",	3))
# print(solution("1010100",	5))
# print(solution("8761639",	3))
print(solution("1769481",	4))

# "1924"	2	"94"
# "1231234"	3	"3234"
# "4177252841"	4	"775841"
# "2635473892"	4	"673892"
# "11111"	4	"1"
# "1080201"	4	"8201"
